test/controller.test.ts

```diff
@@ -56,6 +94,7 @@ const mocksendZclFrameToGroup = vi.fn();
 const mocksendZclFrameToAll = vi.fn();
 const mockAddInstallCode = vi.fn();
 const mocksendZclFrameToEndpoint = vi.fn();
+const mockApaterBackup = vi.fn().mockReturnValue(mockDummyBackup);
```

Comment:
```ts
const mockAdapterBackup = vi.fn(() => Promise.resolve(mockDummyBackup));
```
should provide more accurate typing/testing (and fix typo).
_(A lot of others also need refactoring, but might as well add new ones properly.)_